### PR TITLE
Add rule.resolve

### PR DESCRIPTION
**Pull request by cyjake**
Configuring resolve at rule level is possible [starting from webpack 4.36.1](https://webpack.js.org/configuration/module/#ruleresolve), which is quite useful if same alias name (such as `@`) were shared at different level of packages.

### CODE REVIEW

**Comment by cyjake** [comment]
ping @edmorley

**Comment by opl-** [comment]
Missing documentation in the README and types.

**Comment by cyjake** [comment]
ping @edmorley @opl- Is this pr ready to merge? Plz let me know if there still are any changes needed

**Comment by opl-** [comment]
Personally I'd say it's fine, but I have no power here. Up to @edmorley to review and merge.

**Comment by cyjake** [comment]
ping @edmorley :-/

**Comment by cyjake** [comment]
ping @edmorley :-(

**Comment by edmorley** [comment]
@cyjake Hi! Thank you very much for opening this PR. 

My day to day work no longer involves front-end development, so I don't actually use webpack any more, other than via helping to maintain Neutrino and webpack-chain - as such contributing to these projects is best effort. I can entirely emphasise with opening a PR and having to wait for it to be reviewed/merged - my oldest still-open public PR is from 3 years ago.

However I would like to politely suggest that pinging a PR 5 times in 21 days is too much, and in many cases might actually be counter-productive, since it might annoy maintainers, and mean they are less inclined to use their spare time to take a look. A gentler pinging schedule might be say T+2weeks, T+6weeks, T+3 months etc.

Anyway, I'll take a look at this PR at some point today.

File: test/Rule.js [hunk 1]
```diff
@@ -417,6 +425,29 @@ test('merge with include and exclude not of array type', () => {
...
+  })
+
+  expect(rule.toConfig()).toStrictEqual({
+    resolve: {
+      alias: { foo: 'bar' },
+      extensions: ['.js', '.mjs'],
+    },
+  });
+});
+
```
Comment (cyjake) [inline test/Rule.js:450]: I'm not sure that this assertion covers every case of `merge({ resolve })`. Please give me a nudge if there's still something missing @edmorley

**Comment by edmorley** [comment]
Released in v6.5.0 :-)

**Pull request merged**